src/server/services/wms/qgswmsgetstyles.cpp

```diff
@@ -37,52 +38,33 @@ namespace QgsWms
 
   namespace
   {
-    QDomDocument getStyledLayerDescriptorDocument( QgsServerInterface *serverIface, const QgsProject *project,
-        QStringList &layerList );
+    QDomDocument getStyledLayerDescriptorDocument( QgsServerInterface *serverIface, const QgsProject *project, const QgsServerRequest &request );
   }
 
-  void writeGetStyles( QgsServerInterface *serverIface, const QgsProject *project, const QString &version,
+  void writeGetStyles( QgsServerInterface *serverIface, const QgsProject *project,
                        const QgsServerRequest &request, QgsServerResponse &response )
   {
-    QDomDocument doc = getStyles( serverIface, project, version, request );
+    QDomDocument doc = getStyles( serverIface, project, request );
     response.setHeader( QStringLiteral( "Content-Type" ), QStringLiteral( "text/xml; charset=utf-8" ) );
     response.write( doc.toByteArray() );
   }
 
-  QDomDocument getStyles( QgsServerInterface *serverIface, const QgsProject *project, const QString &version,
+  QDomDocument getStyles( QgsServerInterface *serverIface, const QgsProject *project,
                           const QgsServerRequest &request )
   {
-    Q_UNUSED( version )
-
-    QgsServerRequest::Parameters parameters = request.parameters();
-
-    QString layersName = parameters.value( "LAYERS" );
-
-    if ( layersName.isEmpty() )
-    {
-      throw QgsBadRequestException( QgsServiceException::QGIS_MissingParameterValue,
-                                    QgsWmsParameter::LAYERS );
-    }
-
-#if QT_VERSION < QT_VERSION_CHECK(5, 15, 0)
-    QStringList layerList = layersName.split( ',', QString::SkipEmptyParts );
-#else
-    QStringList layerList = layersName.split( ',', Qt::SkipEmptyParts );
-#endif
-    if ( layerList.isEmpty() )
-    {
-      throw QgsBadRequestException( QgsServiceException::QGIS_MissingParameterValue,
-                                    QgsWmsParameter::LAYERS );
-    }
-
-    return getStyledLayerDescriptorDocument( serverIface, project, layerList );
+    return getStyledLayerDescriptorDocument( serverIface, project, request );
   }
 
   namespace
   {
-    QDomDocument getStyledLayerDescriptorDocument( QgsServerInterface *serverIface, const QgsProject *project,
-        QStringList &layerList )
+    QDomDocument getStyledLayerDescriptorDocument( QgsServerInterface *serverIface, const QgsProject *project, const QgsServerRequest &request )
     {
+      // init WMS parameters and context
+      const QgsWmsParameters parameters( QUrlQuery( request.url() ) );
```

Comment:
@pblottiere 

Why not using directly `const QgsWmsParameters parameters( request.serverParameters() );` instead the implicit conversion from `QUrlQuery` ?

Reply:
Hi @pblottiere 

Looks good to me. See my comment above.

Reply:
> Why not using directly const QgsWmsParameters parameters( request.serverParameters() ); instead the implicit conversion from QUrlQuery ?

Done